Rust 1.80.0 breaks some packages

Dear maintainers,

Rust 1.80.0 contains a couple of changes that break some packages: [an unfixable build regression](https://github.com/rust-lang/rust/issues/127343), and a warning for unrecognized cfg values that some packages treat as an error.  Most commonly, the required fix is just to update the "time" crate.  I've opened PRs to fix as many regressions as I can, but I can't do all of them.

1.80.0 is currently in staging-next, so will probably be in Nixpkgs master in the next 1-2 weeks.  It wasn't really feasible to hold back, because packages tend to start requiring new Rust versions very soon after release, and it also didn't make a lot of sense to keep 1.79.0 as well, because many of the affected packages haven't been touched in a year or more, so there'd be no assurance we'd be able to remove it any time soon.

So, some effort will be required from you to keep your package building with 1.80.0.  Usually, the process will be like this:

- Figure out what's broken.  Usually, but not always, this is that the "time" crate needs to be updated.
- Get it fixed upstream, if it isn't already.
- Ask upstream to release a new version, if they haven't already.
- Update the package in Nixpkgs.

As a last resort, we can patch if we need to, but especially for packages that include a Cargo.lock in Nixpkgs, this is a pain.

I'll post the affected packages in comments, because GitHub limits how many mentions an issue body can contain.

Sorry for the inconvenience.  I've lost a lot of the last week to coordinating the update, collecting broken packages, etc., but hopefully by spreading out the work from here it won't take too much of anybody else's time.

delta is broken:

```
delta>    Compiling ryu v1.0.16
delta>    Compiling time v0.3.31
delta>    Compiling miniz_oxide v0.7.1
delta>    Compiling bstr v1.9.0
delta> error[E0282]: type annotations needed for `Box<_>`
delta>   --> /build/delta-0.17.0-vendor.tar.gz/time/src/format_description/parse/mod.rs:83:9
delta>    |
delta> 83 |     let items = format_items
delta>    |         ^^^^^
delta> ...
delta> 86 |     Ok(items.into())
delta>    |              ---- type must be known at this point
delta>    |
delta> help: consider giving `items` an explicit type, where the placeholders `_` are specified
delta>    |
delta> 83 |     let items: Box<_> = format_items
delta>    |              ++++++++
delta> 
delta>    Compiling line-wrap v0.1.1
delta> For more information about this error, try `rustc --explain E0282`.
delta> error: could not compile `time` (lib) due to 1 previous error
delta> warning: build failed, waiting for other jobs to finish...
aws-sdk-cpp> source root is source
aws-sdk-cpp> Running phase: patchPhase
aws-sdk-cpp> substituteStream() in derivation aws-sdk-cpp-1.11.336: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'cmake/AWSSDKConfig.cmake')
```

--- a/pkgs/by-name/ta/taskchampion-sync-server/package.nix
+++ b/pkgs/by-name/ta/taskchampion-sync-server/package.nix
@@ -5,16 +5,16 @@
 }:
 rustPlatform.buildRustPackage rec {
   pname = "taskchampion-sync-server";
-  version = "0.4.1-unstable-2024-04-08";
+  version = "0.4.1-unstable-2024-08-20";
   src = fetchFromGitHub {
       owner = "GothenburgBitFactory";
       repo = "taskchampion-sync-server";
-      rev = "31cb732f0697208ef9a8d325a79688612087185a";
+      rev = "af918bdf0dea7f7b6e920680c947fc37b37ffffb";
       fetchSubmodules = false;
-      sha256 = "sha256-CUgXJcrCOenbw9ZDFBody5FAvpT1dsZBojJk3wOv9U4=";
+      sha256 = "sha256-BTGD7hZysmOlsT2W+gqj8+Sj6iBN9Jwiyzq5D03PDzM=";
     };
 
-  cargoHash = "sha256-TpShnVQ2eFNLXJzOTlWVaLqT56YkP4zCGCf3yVtNcvI=";
+  cargoHash = "sha256-/HfkE+R8JoNGkCCNQpE/JjGSqPHvjCPnTjOFPCFfJ7A=";
 
   # cargo tests fail when checkType="release" (default)
   checkType = "debug";

